### PR TITLE
Docket Switch | Submit Denial + Denial Success Banner

### LINKED ISSUE #15603: Docket Switch | Submit Denial
## Description
When a user submits the information for denied docket switches (receipt date and context), this will handle 

## Acceptance criteria
- [x] Add Docket Switch controller
- [x] Add create endpoint that we submit the data and disposition to
- [x] Save a `docket_switch` object where the disposition is "denied"
- [x] Save the docket_switch receipt_date on the docket_switch object
- [x] Append the denial context to the existing docket switch denied task instructions
- [x] Close the docket switch denied task
- [x] Redirect to the Case Details page (see Denied Docket switch banner ticket)

### PR DESCRIPTION
**Pull request by wdrougas-tista**
Resolves #15603 

### Description
Please explain the changes you made here.

### Acceptance Criteria
- [ ] Code compiles correctly

### Testing Plan
1. Go to ...

- [ ] For higher-risk changes: [Deploy the custom branch to UAT to test](https://github.com/department-of-veterans-affairs/appeals-deployment/wiki/Applications---Deploy-Custom-Branch-to-UAT)

### User Facing Changes
 - [ ] Screenshots of UI changes added to PR & Original Issue

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] Add or update code comments at the top of the class, module, and/or component.

### Storybook Story
*For Frontend (Presentationa) Components*
* [ ] Add a [Storybook](https://github.com/department-of-veterans-affairs/caseflow/wiki/Documenting-React-Components-with-Storybook) file alongside the component file (e.g. create `MyComponent.stories.js` alongside `MyComponent.jsx`)
* [ ] Give it a title that reflects the component's location within the overall Caseflow hierarchy
* [ ] Write a separate story (within the same file) for each discrete variation of the component

### Database Changes
*Only for Schema Changes*

* [ ] Timestamps (created_at, updated_at) for new tables
* [ ] Column comments updated
* [ ] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [ ] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [ ] Query profiling performed (eyeball Rails log, check bullet and fasterer output)
* [ ] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [ ] DB schema docs updated with `make docs` (after running `make migrate`)
* [ ] #appeals-schema notified with summary and link to this PR
* [ ] Any non-obvious semantics or logic useful for interpreting database data is documented at [Caseflow Data Model and Dictionary](https://github.com/department-of-veterans-affairs/caseflow/wiki/Caseflow-Data-Model-and-Dictionary)

### Integrations: Adding endpoints for external APIs
* [ ] Check that Caseflow's external API code for the endpoint matches the code in the relevant integration repo
  * [ ] Request: Service name, method name, input field names
  * [ ] Response: Check expected data structure
* [ ] Update Fakes
* [ ] Integrations impacting functionality are tested in Caseflow UAT

### CODE REVIEW

**Comment by wdrougas-tista** [comment]
@leikkisa I just connected with @jcq and it looks like we are going to need some clarity on the task statuses for this workflow. The AC in #15471 that focuses on the tasks showing in the Case Timeline conflicts with the current task logic. I'm working to make sure everything passes here and will update #15733 to include this. I can try to find any documentation next week that might clarify this but we'll probably need to ask R&D it sounds like

File: app/helpers/docket_switch_helper.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+module DocketSwitchHelper
+end
```
Comment (leikkisa) [inline app/helpers/docket_switch_helper.rb:4]: This also looks generated and like it can be removed

File: app/models/docket_switch.rb [hunk 2]
```diff
@@ -4,6 +4,9 @@ class DocketSwitch < CaseflowRecord
   belongs_to :old_docket_stream, class_name: "Appeal", optional: false
   belongs_to :new_docket_stream, class_name: "Appeal"
   belongs_to :task, optional: false
+  after_save :process!
```
Comment (leikkisa) [inline app/models/docket_switch.rb:7]: We might want to use after_create, or I'm wondering if we might even want to call this from the controller.

File: spec/requests/docket_switch_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "DocketSwitches", type: :request do
+end
```
Comment (leikkisa) [inline spec/requests/docket_switch_request_spec.rb:6]: This looks generated too (can be removed)

File: spec/feature/queue/docket_switch_spec.rb [hunk 4]
```diff
@@ -190,12 +190,15 @@
...
+      # Redirect to Case Details Page
+      expect(page).to have_current_path("/queue/appeals/#{appeal.uuid}")
 
       # Verify correct success alert
-
+      expect(page).to have_content(format(COPY::DOCKET_SWITCH_DENIAL_SUCCESS_TITLE, appeal.claimant.name))
       # Verify that denial completed correctly
+      expect(docket_switch_denied_task.reload.status).to eq(Constants.TASK_STATUSES.completed)
+      docket_switch = DocketSwitch.find_by(old_docket_stream_id: appeal.id)
+      expect(docket_switch).to_not be_nil
```
Comment (leikkisa) [inline spec/feature/queue/docket_switch_spec.rb:201]: I think it'd be worth checking the instructions too

**Review by leikkisa — COMMENTED** [review]
Just adding some preliminary comments for now, and I need to take another look at the task behavior and conversations we had about that.

**Review by leikkisa — DISMISSED** [review]
LGTM!

**Review by leikkisa — APPROVED** [review]
I thought I approved this already, but I guess not!

**Pull request merged**